airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -699,6 +699,7 @@ def generate_scd_type_2_model(self, from_table: str, column_names: Dict[str, Tup
     {{ sql_table_comment }}
 ),
 {{ '{% endif %}' }}
+{{ '{%- if var("destination") == "clickhouse" %}' }}
```

Comment:
Why do we need this `if` as jinja (runtime) ? instead of doing a if in python and change the generated SQL code when normalization runs (compile)?

Reply:
Because the code block inside the if have some jinja variables, and I don't think there is an option to deep jinja variable attribution. I thought this would be more easy to read. 
```python
file.sql
{{ my_custom_code }}
```

```
s = "{{ another_jinja }}"
render(my_custom_code=s
```

What do you think @ChristopheDuong ? I can move the logic to a python if clause too.

Reply:
Yes, let's move it to python and avoid adding a dbt environment variable for this in the `dbt_project.yml`

Here is a PR to split the jinja templating in multiple rendering:
https://github.com/airbytehq/airbyte/pull/9278